boards/arm/adafruit_itsybitsy_nrf52840/doc/index.rst

```diff
@@ -0,0 +1,153 @@
+following hardware features:
+
++-----------+------------+----------------------+
+| Interface | Controller | Driver/Component     |
++===========+============+======================+
+| ADC       | on-chip    | adc                  |
++-----------+------------+----------------------+
+| CLOCK     | on-chip    | clock_control        |
++-----------+------------+----------------------+
+| FLASH     | on-chip    | flash                |
++-----------+------------+----------------------+
+| GPIO      | on-chip    | gpio                 |
++-----------+------------+----------------------+
+| I2C       | on-chip    | i2c                  |
++-----------+------------+----------------------+
+| MPU       | on-chip    | arch/arm             |
++-----------+------------+----------------------+
+| NVIC      | on-chip    | arch/arm             |
++-----------+------------+----------------------+
+| PWM       | on-chip    | pwm                  |
++-----------+------------+----------------------+
+| RADIO     | on-chip    | Bluetooth,           |
+|           |            | ieee802154           |
++-----------+------------+----------------------+
+| RTC       | on-chip    | system clock         |
++-----------+------------+----------------------+
+| SPI       | on-chip    | spi                  |
++-----------+------------+----------------------+
+| UART      | on-chip    | serial               |
++-----------+------------+----------------------+
+| USB       | on-chip    | usb                  |
++-----------+------------+----------------------+
+| WDT       | on-chip    | watchdog             |
++-----------+------------+----------------------+
+
+Other hardware features have not been enabled yet for this board.
+
+Connections and IOs
+===================
+
+The `Adafruit ItsyBitsy nRF52840 Express Learn site`_ has detailed
+information about the board including `pinouts`_ and the `schematic`_.
+
+LED
+---
+
+* LED0 (red) = P0.06
+* LED1 (Adafruit DotStar)
+    * DATA = P0.08
+    * CLK = P1.09
+
+Push buttons
+------------
+
+* SWITCH = P0.29
+* RESET = P0.18
+
+Testing LEDs and buttons on the Adafruit ItsyBitsy nRF52840 Express
+*******************************************************************
+The :ref:`button-sample` sample lets you test the buttons (switches) and the red LED.
+The :ref:`blinky-sample` sample lets you test the red LED.
+
+The DotStar LED has been implemented as a SPI device and can be tested
+with the :ref:`led_apa102_sample` sample application.
+
+You can build and flash the examples to make sure Zephyr is running correctly on
+your board. The button and LED definitions can be found in
+:zephyr_file:`boards/arm/adafruit_itsybitsy_nrf52840/adafruit_itsybitsy_nrf52840.dts`.
+
+Programming and Debugging
+*************************
+The ItsyBitsy ships with the BOSSA compatible UF2 bootloader.  The
+bootloader can be entered by quickly tapping the reset button twice.
+
+First time setup
```

Comment:
It might be an idea to point out that this guide focuses on the Adafruit Feather but that it does work for the ItsyBitsy